project/plugins/azure.py

```diff
@@ -14,9 +13,7 @@
 def rotate_autoscalers_cloud(configMap, username,  **key_args):
 
     auth = configMap['Global']['azure_credentials'][key_args.get('account')]
-    credentials = ServicePrincipalCredentials(tenant=auth.get('tenant'),
-                                              client_id=auth.get('client_id'),
-                                              secret=auth.get('secret'))
+    credentials = ClientSecretCredential(auth.get('tenant'), auth.get('client_id'), auth.get('secret'))
```

Comment:
Can I safely assume you've tested this change? 🙂

Reply:
Yes. I needed to make this change in order for it to work this cycle. ServicePrincipalCredentials is no longer supported the way we were using it.